internal/error.go

```diff
@@ -119,14 +119,22 @@ Workflow consumers will get an instance of *WorkflowExecutionError. This error w
 */
 
 type (
+	ApplicationErrorOptions struct {
+		NonRetryable   bool
+		Cause          error
+		Details        []interface{}
+		NextRetryDelay time.Duration
+	}
+
 	// ApplicationError returned from activity implementations with message and optional details.
 	ApplicationError struct {
 		temporalError
-		msg          string
-		errType      string
-		nonRetryable bool
-		cause        error
-		details      converter.EncodedValues
+		msg            string
+		errType        string
+		nonRetryable   bool
+		cause          error
+		details        converter.EncodedValues
+		nextRetryDelay time.Duration
```

Comment:
I think we should add a `func (*ApplicationError) NextRetryDelay() time.Duration` akin to the other getters on this struct

Reply:
@cretz these getter functions are only used in tests. When we will need access to the ApplicationError fields outside of the package we can allways add the function.

Reply:
Users may also use those getters

Reply:
@ast2023 - they are used by users too. Just because they are internal, if they are on the struct they are available.